FogCreek/Glitch-Community
# tundra-beat: Two-Factor Authentication

## Links
* Remix link: https://tundra-beat.glitch.me/
* Manuscript link
* Specs/Notion docs: https://www.notion.so/glitch/Authentication-UI-2c0a7259704b486f83c525e9ab33e1d6

## GIF/Screenshots:
![image](https://user-images.githubusercontent.com/8932174/59874962-91e60500-936d-11e9-94d5-3e663655f25a.png)
![image](https://user-images.githubusercontent.com/8932174/59874994-a6c29880-936d-11e9-9d6d-7d51906a0a51.png)
![image](https://user-images.githubusercontent.com/8932174/59875528-02d9ec80-936f-11e9-80d1-3f29ca81d177.png)
![image](https://user-images.githubusercontent.com/8932174/59875495-ea69d200-936e-11e9-953e-017baeaf7726.png)
![image](https://user-images.githubusercontent.com/8932174/59875239-41bb7280-936e-11e9-8300-c2f3cd2f3d27.png)
![image](https://user-images.githubusercontent.com/8932174/59875265-4da73480-936e-11e9-8078-5c4aee52e70d.png)

## Changes:
* Adds "User Passwords" feature flag to `/secret`
* Adds "Account Settings" to User Options popover
* Adds "Set Password" to Account Settings if no password is set
* Adds "Change Password" to Account Settings if a password is set
* Adds "Reset Password" to Account Settings and Sign In popover
* Adds "Enable Two-Factor Authentication" to Account Settings if not enabled
* Adds "Disable Two-Factor Authentication" to Account Settings if enabled
* Adds support for signing in with two-factor authentication enabled

## How To Test:
* Currently, `https://tundra-beat.glitch.me/secret` and enable "User Passwords"
* Check that you can set, update, and reset passwords
* Check that you can enable and disable two-factor authentication

## Feedback I'm looking for:
- ?

## Things left to do before deploying:
- [ ] Review and test?
- [ ] Remove endpoint check for password enabled and use property on user once merged
